lib/Helper.php

```diff
@@ -270,27 +270,120 @@ public static function warn( $message ) {
 		trigger_error( $message, E_USER_WARNING );
 	}
 
+	/**
+	 * Marks something as being incorrectly called.
+	 *
+	 * There is a hook {@see 'doing_it_wrong_run'} that will be called that can be used
+	 * to get the backtrace up to what file and function called the deprecated
+	 * function.
+	 *
+	 * The current behavior is to trigger a user error if `WP_DEBUG` is true.
+	 *
+	 * If you want to catch errors like these in tests, then add the @expectedIncorrectUsage tag.
+	 * E.g.: "@expectedIncorrectUsage Timber::get_posts()".
+	 *
+	 * @api
+	 * @since 2.0.0
+	 * @since WordPress 3.1.0
+	 * @see \_doing_it_wrong()
+	 *
+	 * @param string $function The function that was called.
+	 * @param string $message  A message explaining what has been done incorrectly.
+	 * @param string $version  The version of Timber where the message was added.
+	 */
+	public static function doing_it_wrong( $function, $message, $version ) {
+		/**
+		 * Fires when the given function is being used incorrectly.
+		 *
+		 * @param string $function The function that was called.
+		 * @param string $message  A message explaining what has been done incorrectly.
+		 * @param string $version  The version of WordPress where the message was added.
+		 */
+		do_action( 'doing_it_wrong_run', $function, $message, $version );
+
+		if ( ! WP_DEBUG ) {
+			return;
+		}
+
+		/**
+		 * Filters whether to trigger an error for _doing_it_wrong() calls.
+		 *
+		 * This filter is mainly used by unit tests.
+		 *
+		 * @since WordPress 3.1.0
+		 * @since WordPress 5.1.0 Added the $function, $message and $version parameters.
+		 *
+		 * @param bool   $trigger  Whether to trigger the error for _doing_it_wrong() calls. Default true.
+		 * @param string $function The function that was called.
+		 * @param string $message  A message explaining what has been done incorrectly.
+		 * @param string $version  The version of WordPress where the message was added.
+		 */
+		$should_trigger_error = apply_filters(
+			'doing_it_wrong_trigger_error',
+			true,
+			$function,
+			$message,
+			$version
+		);
+
+		if ( $should_trigger_error ) {
+			if ( is_null( $version ) ) {
+				$version = '';
+			} else {
+				$version = sprintf(
+					'(This message was added in Timber version %s.)',
+					$version
+				);
+			}
+
+			$message .= sprintf(
+				' Please see <a href="%s">Debugging in WordPress</a> as well as <a href="%2$s">Debugging in Timber</a> for more information.',
```

Comment:
Using HTML in an error message will only be useful if the error message is output directly into HTML. However, when we have xDebug enabled, we’ll see raw HTML in the HTML, which is harder to read. Also, when you look at the error message in the console, HTML won’t be parsed either.

That’s why I’d rather have something like this:

```suggestion
				' Please see Debugging in WordPress (%1$s) as well as Debugging in Timber (%2$s) for more information.',
```

Reply:
Agreed.

Reply:
yes! Much nicer